go/client/logclient.go

```diff
@@ -113,6 +124,46 @@ func New(uri string, hc *http.Client) *LogClient {
 	return &LogClient{uri: uri, httpClient: hc}
 }
 
+// NewWithPubKey constructs a new LogClient instance that includes public
+// key information for the log; this instance will check signatures on
+// responses from the log.
+func NewWithPubKey(uri string, hc *http.Client, pemEncodedKey string) (*LogClient, error) {
+	publicBlock, rest := pem.Decode([]byte(pemEncodedKey))
```

Comment:
There's a `PublicKeyFromPEM()` method in the `ct` package, is that helpful?  (You could move your key parameter checks in there)